d3ploy/no_inst.py

```diff
@@ -129,15 +130,28 @@ def print_variables(self):
         print('supply_std_dev: %f' %self.supply_std_dev)
         print('demand_std_dev: %f' %self.demand_std_dev)
 
+    def parse_commodities(self):
+        """ This function parses the vector of strings commodity variable
+            and replaces the variable as a dictionary. This function should be deleted
+            after the map connection is fixed."""
+        temp = copy(self.commodity)
+        self.commodities = {}
+        for entry in self.temp:
+            z = entry.split('_')
+            self.commodities[z[0]].update({[1]: float(z[2])})
```

Comment:
This should just fill out commod_to_fac. Which already has the structure you need I think.

Reply:
If i'm not mistaken, `commod_to_fac` is a dictionary of lists?
in line 155 you create `self.commod_to_fac[commod] = []`.
Am I missing something?

Reply:
the `self.commodities` variable should be a dictionary with:
```
key: commodity name
value: dictionary with:
    key: prototype name
    value: capacity
````

Reply:
So i'm not sure if I can get rid of it cause I thought the capacity the user inputs (i.e. `self.commodities[commod] = {prototype: cap}` ) is an 'expected' capacity, but the `commod_to_fac` was to store the values from the `TIME_SERIES_LISTENERS`?

Reply:
It is, but there is not reason that self.commodities[commod] = {prototype: cap} can't be updated from the time_series_listeners.

Reply:
ah good point. I'll implement that

Reply:
so would the `commodities` variable replace `fac_supply` as well?

Reply:
I think so.